pkg/client/requestor.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright (c) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package client contains the types and functions used to communicate with other services using
+// queues and topics.
+package client
+
+// ResponseCallback is called when a response to a request is received
+// m is the response message
+// requestID is the id of the request this message responds
+type ResponseCallback func(response Message, requestID string) error
```

Comment:
Why is the `requestID` needed here? Can we remove it?

Reply:
The requestID indicates which reuest this message is a response to. we can remove it and add a field to the message (but we said we don't want to change the original message). Note the Responder doesn't know about the id's. Only the Requestor, and that's for his own comfort to track messages. It can also be removed.

Reply:
The `requestID` needs to be part of the message that is written to the requests queue, otherwise the responder can't send back a response to that particular request. What I argued in the meeting is that we should not modify the message object that is passed to the function. Note that the message object in memory and the message written to the queue are two different things.

Also, please consider using `correlationID` for this, it is the typical name for the message field used for this purpose. Some messaging systems even have a built-in `correlationID` property.